docs/contracts/node-operators-registry.md

```diff
@@ -14,126 +39,306 @@ Returns the rewards distribution proportional to the effective stake for each no
 | `used`             | `bool`  | Flag indication if the key was used in the staking    |
 
+### getSigningKeys()
+
+Returns `_limit` signing keys of the node operator.
+
+```sol
+function getSigningKey(uint256 _nodeOperatorId, uint256 _offset, uint256 _limit) returns (
+    bytes pubkeys,
+    bytes signatures,
+    bool[] used
+)
+```
+
+#### Parameters
+
+| Name              | Type      | Description                        |
+| ----------------- | --------- | ---------------------------------- |
+| `_nodeOperatorId` | `uint256` | Node Operator id                   |
+| `_offset`         | `uint256` | Offset of the key, starting with 0 |
+| `_limit`          | `uint256` | Number of keys to return           |
+
+#### Returns
+
+| Name         | Type     | Description                                                                             |
+| ------------ | -------- | --------------------------------------------------------------------------------------- |
+| `pubkeys`    | `bytes`  | Keys concatenated into the bytes batch                                                  |
+| `signatures` | `bytes`  | Signatures concatenated into the bytes batch needed for a deposit_contract.deposit call |
+| `used`       | `bool[]` | Array of flags indicated if the key was used in the staking                             |
+
 ### getNodeOperatorsCount()
 
-Returns total number of node operators
+Returns the total number of node operators.
 
 ```sol
 function getNodeOperatorsCount() returns (uint256)
 ```
 
 ### getKeysOpIndex()
-Returns a monotonically increasing counter that gets incremented when any of the following happens:
-1. a Node Operator's key(s) is added
-2. a Node Operator's key(s) is removed
-3. a Node Operator's approved keys limit is changed
-4. a Node Operator was activated/deactivated
+
+:::note
+DEPRECATED use getNonce() instead
+:::
+
+Returns a counter that MUST change its value whenever the deposit data set changes.
+Below is the typical list of actions that requires an update of the nonce:
+1. a node operator's deposit data is added
+2. a node operator's deposit data is removed
+3. a node operator's ready-to-deposit data size is changed
+4. a node operator was activated/deactivated
+5. a node operator's deposit data is used for the deposit
+
+:::note
+Note: Depending on the StakingModule implementation above list might be extended
+:::
+
+```sol
+function getKeysOpIndex() view returns (uint256)
+```
+
+### getNonce()
+
+Returns a counter that MUST change its value whenever the deposit data set changes.
+Below is the typical list of actions that requires an update of the nonce:
+1. a node operator's deposit data is added;
+2. a node operator's deposit data is removed;
+3. a node operator's ready-to-deposit data size is changed;
+4. a node operator was activated/deactivated;
+5. a node operator's deposit data is used for the deposit.
+
+:::note
+Note: Depending on the StakingModule implementation above list might be extended
```

Comment:
It's already an implementation and should have a particular behavior. Should describe it, not the specification of StakingModule.